kie-wb-common-react/boxed-expression-editor/src/components/Table/Table.tsx

```diff
@@ -0,0 +1,347 @@
+            allowedOperations={tableHandlerAllowedOperations}
+            onOperation={handlingOperation}
+          />
+        }
+      />
+    ),
+    [showTableHandler, handlerConfiguration, tableHandlerAllowedOperations, handlingOperation, tableHandlerTarget]
+  );
+
+  useEffect(() => {
+    onColumnsUpdate(tableColumns.slice(1)); //Removing "# of rows" column
+  }, [onColumnsUpdate, tableColumns]);
+
+  useEffect(() => {
+    onRowsUpdate(tableRows);
+  }, [onRowsUpdate, tableRows]);
+
+  const finishedResizing =
+    tableInstance.state.columnResizing.isResizingColumn === null &&
+    !_.isEmpty(tableInstance.state.columnResizing.columnWidths);
+  useEffect(() => {
+    setTableColumns((prevTableColumns) => {
+      _.forEach(tableInstance.state.columnResizing.columnWidths, (updatedColumnWidth, accessor) => {
+        const columnIndex = _.findIndex(prevTableColumns, { accessor });
+        const updatedColumn = { ...prevTableColumns[columnIndex] };
+        updatedColumn.width = updatedColumnWidth;
+        prevTableColumns.splice(columnIndex, 1, updatedColumn);
+      });
+      return [...prevTableColumns];
+    });
+    // Need to consider a change only when resizing is finished (no other dependencies to consider for this useEffect)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [finishedResizing]);
+
+  return (
+    <div className="table-component">
+      <TableComposable variant="compact" {...tableInstance.getTableProps()}>
+        <Thead noWrap>
+          <tr>
+            {tableInstance.headers.map((column: ColumnInstance, columnIndex: number) =>
+              column.canResize ? (
+                <EditExpressionMenu
+                  title={i18n.editRelation}
+                  selectedExpressionName={column.label}
+                  selectedDataType={column.dataType}
+                  onExpressionUpdate={onColumnNameOrDataTypeUpdate(columnIndex)}
+                  key={columnIndex}
+                >
+                  <Th {...column.getHeaderProps()} {...tableInstance.getThProps(columnIndex)} key={columnIndex}>
+                    <div className="header-cell">
+                      <div>
+                        <p className="pf-u-text-truncate">{column.label}</p>
+                        <p className="pf-u-text-truncate data-type">({column.dataType})</p>
+                      </div>
+                      <div className="pf-c-drawer" {...column.getResizerProps()}>
+                        <div className="pf-c-drawer__splitter pf-m-vertical">
+                          <div className="pf-c-drawer__splitter-handle" />
+                        </div>
+                      </div>
+                    </div>
+                  </Th>
+                </EditExpressionMenu>
+              ) : (
+                <Th {...column.getHeaderProps()} key={columnIndex}>
+                  <div className="header-cell">{column.label}</div>
+                </Th>
+              )
+            )}
```

Comment:
In this kind of scenario, I generally believe it's more interesting to extract each part of the template as a `useCallback` hook (given the else-logic). But, in this scenario is a minor thing, so totally optional ;-)